diff --git a/distributed/worker.py b/distributed/worker.py
index 0f3d8216..53bd8a83 100644
--- a/distributed/worker.py
+++ b/distributed/worker.py
@@ -584,6 +584,7 @@ class Worker(ServerNode):
         dashboard: bool = False,
         http_prefix: str = "/",
         nanny: Nanny | None = None,
+        nanny_contact_address: str | None = None,
         plugins: tuple[WorkerPlugin, ...] = (),
         low_level_profiler: bool | None = None,
         validate: bool | None = None,
@@ -598,6 +599,7 @@ class Worker(ServerNode):
         self.has_what = defaultdict(set)
         self.pending_data_per_worker = defaultdict(deque)
         self.nanny = nanny
+        self.nanny_contact_address = nanny_contact_address
         self._lock = threading.Lock()

         self.data_needed = []
@@ -1171,6 +1173,8 @@ class Worker(ServerNode):
         start = time()
         if self.contact_address is None:
             self.contact_address = self.address
+        if self.nanny_contact_address is None:
+            self.nanny_contact_address = self.nanny
         logger.info("-" * 49)
         while True:
             try:
@@ -1201,7 +1205,7 @@ class Worker(ServerNode):
                         memory_limit=self.memory_limit,
                         local_directory=self.local_directory,
                         services=self.service_ports,
-                        nanny=self.nanny,
+                        nanny=self.nanny_contact_address,
                         pid=os.getpid(),
                         versions=get_versions(),
                         metrics=await self.get_metrics(),
